express-notes-api/server.js

```diff
@@ -0,0 +1,145 @@
+import express from 'express';
+import { readFile, writeFile } from 'node:fs/promises';
+const app = express();
+
+app.use(express.json());
+
+async function readNotes() {
+  try {
+    const data = await readFile('data.json', { encoding: 'utf8' });
+    const jsonData = JSON.parse(data);
+    return jsonData;
+  } catch (err) {
+    console.error(err);
+  }
+}
+
+async function writeNotes(notesData) {
+  try {
+    const jsonData = await readFile('data.json', { encoding: 'utf8' });
+    const data = JSON.parse(jsonData);
+
+    data.notes = notesData.notes;
+    data.nextId = notesData.nextId;
+
+    await writeFile('data.json', JSON.stringify(data, null, 2), { encoding: 'utf8' });
+  } catch (err) {
+    console.error(err);
+  }
+}
```

Comment:
If a file read or write error occurs during a request, the server should respond with a `500 Internal Server Error`. Please add screenshots to your pull request to verify these responses are sent correctly.